src/Umbraco.Core/Hosting/HostingEnvironmentExtensions.cs

```diff
@@ -0,0 +1,33 @@
+using System.IO;
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Core.Hosting;
+
+namespace Umbraco.Extensions
+{
+    public static class HostingEnvironmentExtensions
+    {
+        private static bool? s_isCaseSensitiveFileSystem;
+
+        /// <summary>
+        /// Check to see if the system is a case sensitive filesytem
+        /// </summary>
+        public static bool IsCaseSensitiveFileSystem(this IHostingEnvironment hostingEnv)
+        {
```

Comment:
Could just be `return !RuntimeInformation.IsOSPlatform(OSPlatform.Windows);`

Reply:
Is wondered something like this but I wasn’t sure what other OSs were case insensitive. But yea, I’m sure this would handle most scenarios so might be the simplest option right now 👍🏻

Reply:
Updated as sugested